### PR TITLE
[WIP] Add processing of Warning response

### PR DESCRIPTION
**Pull request by dtrunov**
Add processing of result_code warning from HMI, refactor add_command_request

Related CRQ [APPLINK-29022](https://adc.luxoft.com/jira/browse/APPLINK-29022)

### CODE REVIEW

**Comment by dtrunov** [comment]
@LuxoftAKutsan @tkireva @okozlovlux Please review

**Comment by dtrunov** [comment]
@LuxoftAKutsan @tkireva @okozlovlux Please review latest changes

File: src/components/application_manager/test/commands/mobile/add_command_request_test.cc [hunk 1]
```diff
@@ -129,6 +129,17 @@ class AddCommandRequestTest
     return msg;
   }
 
+  MessageSharedPtr CreateFullParamsSO() {
+    MessageSharedPtr msg = CreateFullParamsUISO();
+    smart_objects::SmartObject& msg_params = (*msg)[am::strings::msg_params];
+    msg_params[strings::vr_commands] =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+    msg_params[strings::vr_commands][0] = "lamer";
+    msg_params[strings::type] = 34;
```
Comment (tkireva) [inline src/components/application_manager/test/commands/mobile/add_command_request_test.cc]: I don't insist, but could you, please, declare this numbers as constants.

**Comment by dtrunov** [comment]
Rebuild required

File: src/components/application_manager/test/commands/mobile/add_command_request_test.cc (hunk 1 above)
Comment (dtrunov) [inline src/components/application_manager/test/commands/mobile/add_command_request_test.cc]: @tkireva  fixed within 886bbfa

Comment (tkireva) [inline src/components/application_manager/test/commands/mobile/add_command_request_test.cc]: @dtrunov OK

**Comment by dtrunov** [comment]
@LuxoftAKutsan Please merge

**Pull request merged**